syntax = "proto3";

package p2p;

message p2pserver {
    oneof commands {
        client_information cl_register = 1;
        client_id cl_lookup = 2;

        client_response response = 10;
    }
}

message client_response {
    oneof data {
        bool response_ok = 1;

        client_information client_info = 10;
    }
}

//generic message
message bool_msg {
    bool success = 1;
}

//generic message
message client_id {
    string name = 1;
}

// The request message containing the user's name.
message client_information {
  client_id id = 1;
  string addr = 2;  // client addr
  uint32 port = 3;  // client binding port (for p2p connections)
}